osgi.specs/docbook/core/003/framework.module.xml

```diff
@@ -2198,18 +2198,29 @@ package-names   ::= package-name                // See <xref
     <section xml:id="i3238802">
       <title>Importing Exported Packages</title>
 
-      <para>Bundles that collaborate require the same class loader for types
-      used in the collaboration. If multiple bundles export packages with
-      collaboration types then they will have to be placed in disjoint
-      class-spaces, making collaboration impossible. Collaboration is
-      significantly improved when bundles are willing to import exported
-      packages; these imports will allow a framework to
-      <emphasis>substitute</emphasis> exports for imports.</para>
-
-      <para>Though substitution is recommended to increase collaboration, it
-      is not always possible. Importing exported packages can only work when
-      those packages are pure API and not encumbered with implementation
-      details. Import of exported packages should only be done when:</para>
+      <para>Bundles that collaborate require the same class loader 
+      for the types used in their interaction. 
+      When multiple bundles export the same package containing collaboration types, 
+      they may end up in separate class-spaces, which can hinder collaboration. 
+      Bundles willing to import exported packages allow the framework to
+      <emphasis>substitute</emphasis> exports for imports and increase collaboration,
+      by trying to avoid separate class-spaces.
+      </para>
+
+      <para>
+      The use of substitutable exports had historical advantages:
+      Implementation bundles embedded and provided the 
+      API packages by OSGi specifications that they implement, 
+      to make themselves easier to deploy. 
+      Instead going forward you should consider not embedding API packages and declare them as dependencies 
+      using valid <code>Import-Package</code> manifest headers with a version range. 
+      At runtime, simply use the API runtime bundles provided by OSGi. 
```

Comment:
It's not just OSGi specifications - we need to be sure people know that it applies to their own code too.

```suggestion
      At runtime the API can be provided by an API bundle, such as the specification API bundles provided by OSGi. 
```